File: SampleAndroid/app/src/main/java/com/vipaar/lime/samplehl/OngoingCallMiniView.kt
package com.vipaar.lime.samplehl

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout

class OngoingCallMiniView(context: Context, attrs: AttributeSet?) : ConstraintLayout(context, attrs) {

    private val label: TextView
        get() = findViewById(R.id.label)
    private val miniViewContainer: FrameLayout
        get() = findViewById(R.id.call_miniview)
    private val maximizeCallButton: View
        get() = findViewById(R.id.button_maximize)
    private val endCallButton: View
        get() = findViewById(R.id.button_end_call)

    var ongoingCallListener: OngoingCallListener? = null

    init {
        inflate(context, R.layout.session_minimized_view, this)
        label.isSelected = true
        maximizeCallButton.setOnClickListener {
            ongoingCallListener?.onMaximizeCall()
        }
        endCallButton.setOnClickListener {
            ongoingCallListener?.onEndCall()
        }
    }

    fun setCallInfo(view: View?, title: String?) {
        if (view == null) {
            visibility = View.GONE
        } else {
            visibility = View.VISIBLE
            label.text = title
            view.parent?.let { parent ->
                (parent as ViewGroup).removeView(view)
            }
            miniViewContainer.removeAllViews()
            miniViewContainer.addView(view)
        }
    }
}

File: SampleXamarin/SampleAndroid/OngoingCallMiniView.cs
using Android.Content;
using Android.Util;
using Android.Views;
using Android.Widget;
using AndroidX.ConstraintLayout.Widget;
using System;

namespace HelpLightning.SDK.Sample.Android
{
    public class OngoingCallMiniView : ConstraintLayout
    {
        private TextView Label => FindViewById<TextView>(Resource.Id.label);
        private FrameLayout MiniViewContainer => FindViewById<FrameLayout>(Resource.Id.call_miniview);
        private View MaximizeCallButton => FindViewById<View>(Resource.Id.button_maximize);
        private View EndCallButton => FindViewById<View>(Resource.Id.button_end_call);

        public IOngoingCallListener OngoingCallListener { get; set; }

        public OngoingCallMiniView(Context context, IAttributeSet attrs) : base(context, attrs)
        {
            Inflate(context, Resource.Layout.session_minimized_view, this);
            Label.Selected = true;
            MaximizeCallButton.Click += (sender, e) => OngoingCallListener?.OnMaximizeCall();
            EndCallButton.Click += (sender, e) => OngoingCallListener?.OnEndCall();
        }

        public void SetCallInfo(View view, string title)
        {
            if (view == null)
            {
                Visibility = ViewStates.Gone;
            }
            else
            {
                Visibility = ViewStates.Visible;
                Label.Text = title;
                if (view.Parent != null)
                {
                    ((ViewGroup)view.Parent).RemoveView(view);
                }
                MiniViewContainer.RemoveAllViews();
                MiniViewContainer.AddView(view);
            }
        }
    }

    public interface IOngoingCallListener
    {
        void OnMaximizeCall();
        void OnEndCall();
    }
}
